promenade/generator.py

```diff
@@ -28,7 +28,7 @@ def validate(self):
                 raise
 
         assert self.input_config['Cluster'].metadata['name'] \
```

Comment:
I believe it would be valuable to wrap this in a try/catch AssertionError and then throw an appropriate error message. Alternatively the assert statement might take a second argument which allows for a custom error message.